### PR TITLE
feat: use event select component select events and filters to preview funnel

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

1. Integrated `EventSelect` component.
2. Modify `EventSelect` component according to actual usage.
3. All funnel request parameters get from frontend component, not MOCK.

## Implementation highlights
截屏2023-08-22 21 18 33

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: frontend/src/ts/const.ts [hunk 1]
```diff
@@ -411,9 +411,46 @@ export enum MetadataPlatform {
...
   STRING = 'String',
-  NUMBER = 'Number',
-  BOOLEAN = 'Boolean',
-  DATETIME = 'DateTime',
+  INTEGER = 'Integer',
+  DOUBLE = 'Double',
+  FLOAT = 'Float',
+}
+
+export enum MetadataParameterType {
```
Comment (zxkane) [inline frontend/src/ts/const.ts:419]: should we share those models in frontend and backend projects?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: frontend/src/ts/const.ts (hunk 1 above)
Comment (tyyzqmf) [reply frontend/src/ts/const.ts:419]: link file can't loader by frontend with **enum** types.

File: frontend/src/pages/analytics/funnel/AnalyticsFunnel.tsx [hunk 2]
```diff
@@ -294,68 +666,322 @@ const AnalyticsFunnel: React.FC = () => {
...
+                        key: 'previous-1-year',
+                        amount: 1,
+                        unit: 'year',
+                        type: 'relative',
+                      },
+                    ]}
+                    isValidRange={(
+                      range: DateRangePickerProps.Value | null
+                    ) => {
+                      if (range?.type === 'absolute') {
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/funnel/AnalyticsFunnel.tsx:731]: Too complex. Replace this logic as a function/method.

File: frontend/src/ts/utils.ts [hunk 3]
```diff
@@ -312,3 +320,122 @@ export const getValueFromStackOutputs = (
   }
   return res;
 };
+
+export const metadataEventsConvertToCategoryItemType = (
+  apiDataItems: IMetadataEvent[]
+) => {
+  const categoryItems: CategoryItemType[] = [];
+  const categoryPresetItems: CategoryItemType = {
+    categoryName: '预置事件',
```
Comment (chenhaiyun) [inline frontend/src/ts/utils.ts:329]: change to i18n

File: frontend/src/ts/utils.ts [hunk 4]
```diff
@@ -312,3 +320,122 @@ export const getValueFromStackOutputs = (
...
+};
+
+export const parametersConvertToCategoryItemType = (
+  userAttributeItems: IMetadataUserAttribute[],
+  parameterItems?: IMetadataEventParameter[],
+  relationItems?: IMetadataRelation[]
+) => {
+  const categoryItems: CategoryItemType[] = [];
+  const categoryEventItems: CategoryItemType = {
+    categoryName: '事件属性',
```
Comment (chenhaiyun) [inline frontend/src/ts/utils.ts:369]: change to i18n

File: frontend/src/components/eventselect/AnalyticsType.ts [hunk 5]
```diff
@@ -111,101 +89,64 @@ export const DEFAULT_SEGMENTATION_DATA: IConditionItemType = {
...
-  { value: 'not_match', label: '不匹配正则' },
-];
+export const ANALYTICS_OPERATORS = {
+  is_null: { value: 'is_null', label: 'null' },
+  is_not_null: { value: 'is_not_null', label: '非null' },
+  equal: { value: '=', label: '=' },
+  not_equal: { value: '<>', label: '!=' },
+  not_equal_contain_null: {
+    value: 'not_equal_contain_null',
+    label: '!=(含null)',
```
[the remark this answers is not in the document]
Comment (tyyzqmf) [reply frontend/src/components/eventselect/AnalyticsType.ts:104]: Nice! This is exactly what I need.

**Review by zxkane — APPROVED** [review]

**Pull request merged**